src/db_type/key/key.rs

```diff
@@ -343,7 +353,11 @@ impl KeyRange {
                 start: start.to_key(),
             }),
             (Bound::Excluded(start), Bound::Included(end)) => {
-                KeyRange::RangeInclusive(start.to_key()..=end.to_key())
+                let start = start.to_key();
+                let mut end = end.to_key();
+                // Add 255 to the end key to include the last key
+                end.extend(&Key::new(vec![255]));
```

Comment:
There is no difference between this and the one above? Should this be `start.extend` ?

Reply:
Well spotted, you're absolutely right to find this code strange; it's one of the things that needs refactoring and will be removed.


By the way, just a little background on this legacy code: in the case of Rust ranges and the workarounds put in place here, it’s all about the inclusive end range because we use `start..=end`.

Reply:
Here is the cleaned code https://github.com/vincent-herlemont/native_db/commit/a0b2e8cf1a65ce298cdcec31347445bd4af04643, in fact, part of the code might not be supported due to the syntax of [ranges in Rust](https://doc.rust-lang.org/std/ops/struct.Range.html).